### PR TITLE
Check for response status in Client.Store() method

### PR DESCRIPTION
**Pull request by chetanyakan**
**Summary**

* [x] Check for response.StatusCode in Client.Store() method

Explain the **motivation** for making this change. What existing problem does the pull request solve?

- When converting an office file to PDF, sometimes when the conversion fails, a file of ~57B is written as the result PDF. I would expect gotenberg-go-client to return an error in this case. 

**Test plan (required)**

Demonstrate the code is solid. Example: The exact commands you ran and their output.

**Closing issues**

Fixes #

**Checklist**

- [ ] Have you followed the guidelines in our [CONTRIBUTING](CONTRIBUTING.md) guide?
- [ ] Have you lint your code locally prior to submission (`make lint`)?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally (`make tests`)?
- [ ] I have squashed any insignificant commits
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -110,6 +110,10 @@ func (c *Client) Store(req Request, dest string) error {
 	if err != nil {
 		return err
 	}
+	// Check for 2XX Status Codes
+	if resp.StatusCode < http.StatusOK || resp.StatusCode >= http.StatusMultipleChoices {
```
Comment (harshilsharma63) [inline client.go]: Does this function `Client.Store()` only get called for conversion to PDFs?

File: client.go [hunk 2]
```diff
@@ -110,6 +110,10 @@ func (c *Client) Store(req Request, dest string) error {
...
+		return errors.New("cannot convert file to PDF")
```
Comment (harshilsharma63) [inline client.go]: `couldn't` seems to be better than `cannot` as we do not know if the file can never be converted.

**Review by harshilsharma63 — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (chetanyakan) [reply client.go]: `Client.Store()` only gets called for the conversion to PDFs or for merging several PDF files. I've updated the error message returned to reflect the same.

**Review by harshilsharma63 — APPROVED** [review]

**Comment by chetanyakan** [comment]
Closing as changes were merged in the fork

**Pull request closed without merging**